src/Toml/Bi/Combinators.hs

```diff
@@ -162,6 +163,11 @@ string = match _String
 read :: (Show a, Read a) => Key -> TomlCodec a
 read = match _Read
 
+-- | Codec for general nullary sum data types with a 'Bounded', 'Enum', 'Read',
+-- and 'Show' instance.
+sumType :: (Bounded a, Enum a, Read a, Show a) => Key -> TomlCodec a
+sumType = match _SumType
```

Comment:
`sumType` is a quite generic name. This codec doesn't support general sum types. So let's rename to `enumBounded` so the name reflects the idea. And also let's rename `_SumType` to `_EnumBounded`.